src/components/leaflet/sy-marker.tsx

```diff
@@ -0,0 +1,26 @@
+import SyMarkerImage from './sy-marker.png';
+import Leaflet from 'leaflet';
+
+const isBrowser = typeof window !== 'undefined';
+
+/**
+ * Leaflet doesn't support SSR, so we run into issues
+ * with window if leaflet is loaded directly. Check file gatsby-node
+ * for the webpack config injection with replaces it with the null loaders.
+ * This also means Leaflet will be null here in this case. Handle accordingly.
+ *
+ * Related:
+ * https://github.com/PaulLeCam/react-leaflet/issues/495
+ */
+let SatellytesMarkerIcon: any = undefined;
+
+if (isBrowser) {
```

Comment:
When excluding components from the SSR, the `isBrowser` check should be done by using `useEffect`. 

```
const [isBrowser, setIsBrowser] = useState(false);
useEffect(() => {
	setIsBrowser(true)
})
```

The reason for this is, that during hydration the result must be the same on the first render. If you put the browser check outside of a react, this is not the case anymore and you get hydration errors in the console.

I checked the code on my machine, as Github doesn't show all changes:
- for the `SatellytesMarkerIcon` I guess the `window !== 'undefined'` check is ok, as the `Leaflet` component also has a browser check
- the leaflet component should use the `useEffect` check, for optimal hydration

Reply:
thank you for the useEffect hint. What's the best way to test the outcome? Do I always have to build or is it enough to disable javascript? Or as you commonly demonstrate, check the network pane for the html payload ? 

I will work on the Leaflet component to make SSR better here.

Reply:
Most of the time, I just check the outcome in the network tab. Disabling JS is a good idea, never tried it.